pkg/agent/core/ngt/service/ngt.go

```diff
@@ -170,12 +170,26 @@ func (n *ngt) initNGT(opts ...core.Option) (err error) {
 		n.core, err = core.New(opts...)
 		return err
 	}
+	if os.IsPermission(err) {
+		log.Debugf("no permission for index path,\tpath: %s,\terr: %v", n.path, err)
+		return err
+	}
 
 	log.Debugf("load index from %s", n.path)
 
 	agentMetadata, err := metadata.Load(filepath.Join(n.path, metadata.AgentMetadataFileName))
 	if err != nil {
 		log.Warnf("cannot read metadata from %s: %s", metadata.AgentMetadataFileName, err)
+
```

Comment:
I think we should check metadata.Load's err is os.IsNotExists or not, if err is os.ErrNotExists we can check kvsdb exsitance.